packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -69,48 +77,82 @@ class WidgetFlutterBinding extends BindingBase with Scheduler, Gesturer, Service
     );
   }
 
-  /// The one static instance of this class.
-  ///
-  /// Only valid after the WidgetFlutterBinding constructor) has been called.
-  /// Only one binding class can be instantiated per process. If another
-  /// BindingBase implementation has been instantiated before this one (e.g.
-  /// bindings from other frameworks based on the Flutter "rendering" library),
-  /// then WidgetFlutterBinding.instance will not be valid (and will throw in
-  /// checked mode).
-  static WidgetFlutterBinding get instance => _instance;
-  static WidgetFlutterBinding _instance;
+  /// The [BuildOwner] in charge of executing the build pipeline for the
+  /// widget tree rooted at this binding.
+  BuildOwner get buildOwner => _buildOwner;
+  final BuildOwner _buildOwner = new BuildOwner();
 
-  final List<BindingObserver> _observers = new List<BindingObserver>();
+  final List<WidgetsBindingObserver> _observers = <WidgetsBindingObserver>[];
 
-  void addObserver(BindingObserver observer) => _observers.add(observer);
-  bool removeObserver(BindingObserver observer) => _observers.remove(observer);
+  /// Registers the given object as a binding observer. Binding
+  /// observers are notified when various application events occur,
+  /// for example when the system locale changes. Generally, one
+  /// widget in the widget tree registers itself as a binding
+  /// observer, and converts the system state into inherited widgets.
+  ///
+  /// For example, the [WidgetsApp] widget registers as a binding
+  /// observer and converts the screen size into [MediaQuery] widgets.
```

Comment:
By "widgets" I think you mean MediaQuery and MediaQueryData. It might be enough to just say MediaQuery here.